extension/source/QuillPage/CellsDemo/BalanceWidget.tsx

```diff
@@ -0,0 +1,69 @@
+import { ethers } from 'ethers';
+import { FunctionComponent, useMemo } from 'react';
+import { FormulaCell } from '../../cells/FormulaCell';
+import MemoryCell from '../../cells/MemoryCell';
+import { useQuill } from '../QuillContext';
+import { Display } from './Display';
+import TextBox from './TextBox';
+
+const BalanceWidget: FunctionComponent = () => {
+  const quillCtx = useQuill();
+
+  const cells = useMemo(() => {
+    const address = new MemoryCell('');
+
+    const balanceDisplay = new FormulaCell(
+      { address, _: quillCtx.blockNumber },
+      // eslint-disable-next-line @typescript-eslint/no-shadow
+      async ({ address }) => {
```

Comment:
I really, really don't like having these disables everywhere. The [no-shadow](https://eslint.org/docs/rules/no-shadow) prevents programmatic/logical errors, so I would like to leave it enabled.

I would just rename the shadows, or consider using array destructuring simlar to `React.useState`
```suggestion
      async ({ address: addr }) => {
```

Reply:
Yeah I touched on this in the video. The awkward thing is that this will lead to artificial renames/synonyms similar to your suggestion, which is also its own antipattern.

I have an idea. Will follow-up.

Reply:
Here's my proposed fix: https://github.com/web3well/bls-wallet/pull/239.

In summary, it uses [template literal types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) to allow you to write this in a type-checked way:

```ts
const ab = new FormulaCell({ a, b }, ({ $a, $b }) => $a * $b);
```